libkbfs/folder_block_ops.go

```diff
@@ -1824,10 +1825,24 @@ func (fbo *folderBlockOps) maybeWaitOnDeferredWrites(
 				"Blocking a write because of a full dirty buffer")
 			doLogUnblocked = true
 		case err := <-errListener:
-			// XXX: should we ignore non-fatal errors (like
-			// context.Canceled), or errors that are specific only to
-			// some other file being sync'd (e.g., "recoverable" block
-			// errors from which we couldn't recover)?
+			// Context errors are safe to ignore, since they are
+			// likely to be specific to a previous sync (e.g., a user
+			// hit ctrl-c during an fsync, or a sync timed out, or a
+			// test was provoking an error specifically [KBFS-2164]).
+			if err == context.Canceled || err == context.DeadlineExceeded {
+				fbo.log.CDebugf(ctx, "Ignoring sync err: %+v", err)
+				err := registerErr()
+				if err != nil {
+					return err
```

Comment:
Just to make sure a reader sees the comment below before, I'd suggest a refactor to the following:
```go
if ...
    fbo.log..CDebugf(...)
    err = registerErr()
    if err == nil {
        continue
    }
}
// Treat other errors...
return err
```